components/root/root.jsx

```diff
@@ -146,7 +146,7 @@ export default class Root extends React.Component {
         const diagnosticId = this.props.diagnosticId;
 
         /*eslint-disable */
-        if (segmentKey != null && segmentKey !== '' && this.props.diagnosticsEnabled) {
+        if (segmentKey != null && segmentKey !== '' && !segmentKey.startsWith('placeholder') && this.props.diagnosticsEnabled) {
```

Comment:
I don't understand the purpose of this code.

The segmentKey is obtained from a constant, so, It'll never be null, and maybe, It'll be an empty string (if somebody modify the code to "disable" segment removing the key), but then, the placeholder approach is "redundant" and a bit confusing for me. For me is easier to set the constant to an empty string (so the purpose is clear). Anyway, it is a constant, so we can remove the entire block, or move it to a config setting in the server.

I'm ok on merging this as is, because I understand the idea behind, but I think is a confusing way to write it.

Reply:
It's done this way because Jenkins runs a `sed s/placeholder_segment_key/$real_segment_key/g` over the codebase, to update that constant to an appropriate value depending on the build type.

Reply:
ok, that sounds sensible

Reply:
Worth an inline command as such?